### DISCUSSION TITLE
Waveshare Pi Pico Eval board + ESP32-S3

### DISCUSSION THREAD

**Discussion by PJ5168** [Bodmer/TFT_eSPI#2357] [General]
Hi,

The Pi Pico works on the Waveshare Pi Pico Eval board with RPI_DISPLAY_TYPE setting.
The ESP32-S3 does NOT work on Waveshare Pi Pico Eval board with RPI_DISPLAY_TYPE setting.

The Waveshare Pi Pico Eval board used ILI9488 driver with serial SPI to parallel TTLs.
And the Eval board uses the same SPI pins on Waveshare Pi Pico 1.3" HAT. 
ESP-S3 works on the 1.3" HAT. Therefore, the ESP32-S3 shall be able to drive those SPI pins on Eval board.

Please help if any. Appreciate!
Note: ESP32-S3 board used is Banana Pi-PicoW-S3: https://wiki.banana-pi.org/BPI-PicoW-S3

Best,

PJ.

**Comment by Bodmer** [comment]
The different board will require a new setup file with the correctly mapped pins.

**Comment by Bodmer** [comment]
Looking at the product data it looks like this is the correct maping for the two processors:

RP2040
#define TFT_MOSI 11
#define TFT_SCLK 10
#define TFT_MISO 12

#define TFT_CS   9
#define TFT_DC   8
#define TFT_RST 15

S3
#define TFT_MOSI 38
#define TFT_SCLK 21
#define TFT_MISO 39

#define TFT_CS  16
#define TFT_DC  18
#define TFT_RST 42

**Reply by PJ5168** [reply]
Dear Sir,

Yes, used the same pins as your comments.

// ESP32 PicoW-S3 with 3.5" Pico Eval board 480x320 ILI9488
//                                   S3 GPIO   Pi Pico GP 
// #define TFT_MISO     39            // 12
// #define TFT_MOSI     38            // 11
// #define TFT_SCLK      21           // 10
// #define TFT_CS          16           // 9
// #define TFT_DC         18          // 8
// #define TFT_RST        42          // 15
// #define TFT_BL          40          // 13 HIGH ON
// #define TP_CS             1          // 16
// #define TP_IRQ           2          // 17

Test results :
RP2040 + Eval board    = LCD and touch OK (with RPI_DISPLAY_TYPE)
RP2040 + 1.3" HAT       = LCD OK
ESP32-S3 + 1.3" HAT    = LCD OK
ESP32-S3 + Eval board = LCD fail (with RPI_DISPLAY_TYPE)

The Eval board and 1.3" HAT use same SPI pins. So, ESP32-S3 can drive those SPI pins.

Look at the .h file in the processors directory, the RP2040 is different from ESP32-S3.
RP2040:
    #elif  defined (RPI_DISPLAY_TYPE) // RPi TFT type always needs 16 bit transfers
      #define tft_Write_8(C)   spi.transfer(C); spi.transfer(C)
      #define tft_Write_16(C)  spi.transfer((uint8_t)((C)>>8));spi.transfer((uint8_t)((C)>>0))
      #define tft_Write_16N(C) spi.transfer((uint8_t)((C)>>8));spi.transfer((uint8_t)((C)>>0))
      #define tft_Write_16S(C) spi.transfer((uint8_t)((C)>>0));spi.transfer((uint8_t)((C)>>8))

Thank you.

PJ.

**Reply by Frank-Odirile** [reply]
Greetings, Please can you share some sample code for 3.5" Pico Eval board 480x320 ILI9488 using Rasbery Pico.

**Comment by PJ5168** [comment]
ESP32-S3 shows image on the LCD now using the example code from Waveshare instead.

**Comment by PJ5168** [comment]
And the touch is working as well.

**Comment by PJ5168** [comment]
SPI MODE3, MSB first
//-------------------------------------------------- 
void LCD_WriteReg(uint8_t Reg)
{
    LCD_DC_0;
    LCD_CS_0;
    SPI4W_Write_Byte(Reg);
    LCD_CS_1;
}
//-------------------------------------------------
void LCD_WriteData(uint16_t Data)
{
    LCD_DC_1;
    LCD_CS_0;
    SPI4W_Write_Byte(Data >> 8);
    SPI4W_Write_Byte(Data & 0XFF); //
    LCD_CS_1;
}
//-------------------------------------------------
#define SPI4W_Write_Byte(__DATA) SPI.transfer(__DATA)

**Comment by Bodmer** [comment]
I had a look at the code and see that RPi stle interface display support for the ESP32 S3 had not been tested and the original ESP32 code was in place.

I have updated the library so all should be good now.

**Comment by PJ5168** [comment]
Dear Sir,
Appreciate for the update.
The new library is working up to 20Mhz. The touch function is also works. 
Thanks again for the great library!
Best,
PJ
